curveball/models.py

```diff
@@ -28,11 +30,38 @@ class CurveballBlock(models.Model):
     curveball_three = models.ForeignKey(Curveball, null=True, blank=True,
                                         related_name='curveball_three')
 
+    def _get_section(self):
+        return self.pageblock().section
+
+    section = property(_get_section)
+
     def __unicode__(self):
         return unicode(self.pageblock())
 
+    def get_curveballs(self):
+        return [self.curveball_one, self.curveball_two, self.curveball_three]
+
+    def get_latest_curveball_submission(self, group_user):
+        selected_curveballs = []
+        cbs = self.get_curveballs()
+
+        for cb in cbs:
+            try:
+                cb_sub = cb.curveballsubmission_set.filter(
+                    group_curveball=group_user).latest('submitted')
+                if cb_sub:
+                    selected_curveballs.append(cb_sub)
+            except:
```

Comment:
bare `except`s are bad form in most cases. If there's a particular exception that may be raised in there, handle it explicitly.